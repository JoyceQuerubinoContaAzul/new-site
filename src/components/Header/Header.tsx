// import Image from 'next/image';
import { cn } from '@/utils/cn';
import { Container } from '@/components/Container';
import type { HeaderProps } from './Header.types';

export const Header = (props: HeaderProps) => {
  const { className, noLoginAction, ...rest } = props;

  return (
    <header
      className={cn(
        'sticky top-0 z-50 bg-white py-3.5 md:static md:pt-[46px] md:pb-7 dark:bg-[#1c191a]',
        className,
      )}
      {...rest}
    >
      <Container className="flex items-center justify-between gap-3">
        <div className="h-[24px] w-[167px]">
          <img src="/images/ca-logo.svg" alt="Logo" />
        </div>

        {!noLoginAction && (
          <div className="md:hidden">
            <p className="text-right text-xs font-medium">
              JÃ¡ tem uma conta?
            </p>
          </div>
        )}
      </Container>
    </header>
  );
};
