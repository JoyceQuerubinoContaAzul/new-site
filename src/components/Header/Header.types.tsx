export type HeaderProps = React.ComponentProps<'header'> & {
  noLoginAction?: boolean;
};
