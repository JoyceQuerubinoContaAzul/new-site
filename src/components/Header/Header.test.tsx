import { render } from '@/tests';
import { Header } from './Header';

type HeaderProps = React.ComponentProps<typeof Header>;

const setup = (props: Partial<HeaderProps> = {}) =>
  render(<Header {...props} />);

describe('Header component', () => {
  describe('when the "noLoginAction" is true', () => {
    it('should not render login link', () => {
      const { queryByRole } = setup({ noLoginAction: true });

      expect(queryByRole('link', { name: 'Entrar' })).not.toBeInTheDocument();
    });
  });

  describe('when the "noLoginAction" is falsy', () => {
    it('should render login link', () => {
      const { getByRole } = setup({ noLoginAction: false });

      expect(getByRole('link', { name: 'Entrar' })).toBeInTheDocument();
    });
  });

  it('should render the entire header', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
