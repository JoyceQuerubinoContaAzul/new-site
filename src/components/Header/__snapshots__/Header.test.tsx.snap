// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render the entire header 1`] = `
<div>
  <header
    class="sticky top-0 z-50 bg-white py-3.5 md:static md:pt-[46px] md:pb-7 dark:bg-[#1c191a]"
  >
    <div
      class="mx-auto max-w-[520px] px-5 flex items-center justify-between gap-3"
    >
      <div
        class="relative h-[17px] w-[130px] md:h-[33px] md:w-[250px]"
      >
        <img
          alt="Conta Azul"
          class="object-contain"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="md:hidden"
      >
        <p
          class="text-right text-xs font-medium"
        >
          JÃ¡ tem uma conta? 
          <a
            class="text-blue-ca-semi-dark underline"
            href="https://login.contaazul.com"
          >
            Entrar
          </a>
        </p>
      </div>
    </div>
  </header>
</div>
`;
